### PR TITLE
Fix/rest client jaxrs warning

### LINKED ISSUE #2667: Missing JaxRS beans in the Jandex index
**Describe the bug**
When returning a `javax.ws.rs.core.Response` in a RestClient, I have the following WARNING in the log, there seems to be some misisng beans in the Jandex index .

```
2019-05-31 09:56:58,610 WARN  [io.qua.dep.ste.ReflectiveHierarchyStep] (build-2) Unable to properly register the hierarchy of the following classes for reflection as they are not in the Jandex index:
	- javax.ws.rs.client.ClientResponseContext
	- javax.ws.rs.core.EntityTag
	- javax.ws.rs.core.Link
	- javax.ws.rs.core.MediaType
	- javax.ws.rs.core.NewCookie
	- javax.ws.rs.core.Response
	- javax.ws.rs.core.Response$StatusType
Consider adding them to the index either by creating a Jandex index for your dependency via the Maven plugin, an empty META-INF/beans.xml or quarkus.index-dependency properties.");.
```
A returning a Response object is a very common thing (when using POST to create an entity for exemple), these should be included in the Jandex index by default. I check the code in SmallRyeRestClientProcessor::processInterfaces and the return types seems to be registered so I'm not sure it's a real issue a just a log issue. And as it works maybe it's just a log issue ;)

**Expected behavior**
No warning in the log file.

**To Reproduce**
1. Create an application following the Rest client tutorial : https://quarkus.io/guides/rest-client-guide
2. Add a method using a remote POST operation and returning a Response object to the CountriesService interface.

Using these simple steps from the tutorial, we also seen a lot a warning from Resteasy saying that some providers are already registred and so the second registration will be ignored, they seems to be ignorable but it's still ugly to have them in the log.

**Environment (please complete the following information):**
 - Quarkus version or git rev: 0.15.0

### PR DESCRIPTION
**Pull request by loicmathieu**
This fixes #2667, as adviced by @gsmet I extract the constant TYPES_IGNORED_FOR_REFLECTION in resteasy-common to be able to share it between smallrye rest client and resteasy server.
I choose the constant interface ResteasyDotNames for this purpose.

Tested a few minutes ago on master.

### CODE REVIEW

File: extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java [hunk 1]
```diff
@@ -39,4 +45,24 @@
...
 
+    //types ignored for reflection used by resteasy and smallrye rest client.
+    public static final Set<DotName> TYPES_IGNORED_FOR_REFLECTION = new HashSet<>(Arrays.asList(
+            // javax.json
+            DotName.createSimple("javax.json.JsonObject"),
+            DotName.createSimple("javax.json.JsonArray"),
+
+            // JAX-RS
+            DotName.createSimple(Response.class.getName()),
+            DotName.createSimple(AsyncResponse.class.getName()),
```
Comment (gsmet) [inline extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java]: I know it was preexisting but could you use the class names as strings instead? There's no need to load the classes if they are not used (and they might as well not be).

File: extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java [hunk 2]
```diff
@@ -39,4 +45,24 @@
     public static final DotName RESTEASY_MATRIX_PARAM = DotName
             .createSimple(org.jboss.resteasy.annotations.jaxrs.MatrixParam.class.getName());
 
+    //types ignored for reflection used by resteasy and smallrye rest client.
```
Comment (gsmet) [inline extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java]:
```suggestion
    // Types ignored for reflection used by the RESTEasy and SmallRye REST client extensions.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks good overall. I added 2 minor comments.

Could you take care of them and squash everything into 1 commit? We don't need 4 of them for that change :)

Thanks!

File: extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java (hunk 1 above)
Comment (loicmathieu) [reply extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java]: Fixed

**Comment by gsmet** [comment]
@loicmathieu thanks! Could you also get rid of the 2 merge commits?

**Pull request closed without merging**